### PR TITLE
Fix issue of having spaces in the node name when saving data to a h5 …

### PR DESCRIPTION
**Pull request by knoxsp**
…file

### CODE REVIEW

File: hydra_pywr/runner.py [hunk 1]
```diff
@@ -677,15 +678,16 @@ def generate_array_recorder_resource_scenarios(self):
                 resultstore = pandas.HDFStore(os.path.join(self.results_location, filename), mode='w')
                 self.resultstores[filename] = resultstore
 
-            resultstore.put(f"{nodename}", df)
-            resultstore[f"{nodename}"].attrs['pandas_type'] = 'frame'
+            noderef = re.sub(r'[^a-zA-Z0-9]', '', nodename)
```
Comment (pmslavin) [inline hydra_pywr/runner.py:681]: According to the [regex in PyTables](https://github.com/PyTables/PyTables/blob/5a0386a20a2e6bce5d1231dcfd9556d7034e5e40/tables/path.py#L21C1-L21C55) digits are not permitted as initial characters of names but underscores are allowed to appear anywhere, so the replacement here still allows some `NaturalNameWarning` to be generated and removes underscores unecessarily.

The following removes a series of unwanted chars and 0-9 from the start of the name, then any unwanted chars from the remainder.

```suggestion
            noderef = re.sub(r'^[^a-zA-Z_]+|[^a-zA-Z0-9_]', '', nodename)
```

**Review by pmslavin — COMMENTED** [review]
One suggestion to prevent some NaturalNameWarning and allow underscores in nodenames.

**Comment by knoxsp** [comment]
Thanks for the suggestion. Applied.

**Review by pmslavin — APPROVED** [review]
Should now disallow all nodenames that would lead to PyTables NaturalNameWarning.

**Pull request merged**